### PR TITLE
Clean up users.bio

### PR DESCRIPTION
**Pull request by eviljeff**
Separate migration because we need to deploy separately.
Follow on from https://github.com/mozilla/addons-server/pull/5279

### CODE REVIEW

**Comment by diox** [comment]
I wonder if we should try handling the different FK names - we've done this in various other migrations in the past and it's super painful, but also super helpful for others on the team who don't want to deal with that mess...

Locally, my FK is named `bio_refs_id_b985d533`. Is yours the same?

**Comment by eviljeff** [comment]
I already dropped mine during development of the patch, alas.

File: src/olympia/migrations/939-drop-l10n-from-bio-user-profile-cleanup.sql [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+DELIMITER ';;'
+
+CREATE PROCEDURE drop_bio_fk() BEGIN
+    /* drop the bio foreign key so we can clear the translations.
+    - we know prod and local dbs have different keys, so try to handle both */
+
+    /* prod fk name */
+    IF EXISTS (SELECT 1 FROM information_schema.table_constraints WHERE table_schema = (SELECT DATABASE()) AND table_name = 'users' AND constraint_name = 'users_ibfk_1') THEN
+        ALTER TABLE reviews DROP FOREIGN KEY users_ibfk_1;
```
Comment (diox) [inline src/olympia/migrations/939-drop-l10n-from-bio-user-profile-cleanup.sql]: Wrong copy paste, you have the right `table_name` in the `SELECT` but the wrong one in the `ALTER`

File: src/olympia/migrations/939-drop-l10n-from-bio-user-profile-cleanup.sql [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    END IF;
+
+    /* local docker default fk name */
+    IF EXISTS (SELECT 1 FROM information_schema.table_constraints WHERE table_schema = (SELECT DATABASE()) AND table_name = 'users' AND constraint_name = 'users_bio_21cc3d1ebf26e19_fk_translations_id') THEN
+        ALTER TABLE reviews DROP FOREIGN KEY users_bio_21cc3d1ebf26e19_fk_translations_id;
```
Comment (diox) [inline src/olympia/migrations/939-drop-l10n-from-bio-user-profile-cleanup.sql]: Same issue

Comment (diox) [inline src/olympia/migrations/939-drop-l10n-from-bio-user-profile-cleanup.sql]: Looks like `src/olympia/migrations/823-drop-backup-version.sql` is a better source of inspiration anyway. It allows us to retrieve the constraint name in one query which is much easier to read. So I'd replace the entire procedure with:

```sql
SET @CONSTRAINT_NAME := (SELECT CONSTRAINT_NAME FROM information_schema.key_column_usage WHERE TABLE_SCHEMA="olympia" AND TABLE_NAME="users" and COLUMN_NAME="bio" AND REFERENCED_TABLE_NAME = "translations");

SET @QUERY = CONCAT('ALTER TABLE users DROP FOREIGN KEY ', @constraint_name, ';');
PREPARE stmt FROM @QUERY;
EXECUTE stmt;
DEALLOCATE PREPARE stmt;
```

Comment (eviljeff) [inline src/olympia/migrations/939-drop-l10n-from-bio-user-profile-cleanup.sql]: https://github.com/mozilla/addons-server/pull/5316/commits/2e489ea7d0450ca97558039b72a1e34f2fd913a6

**Pull request merged**